spec/services/group_notifier_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+describe GroupNotifier do
+  let(:current_user) { double(User, name: 'user', id: 1) }
+  let(:group) { double(Group, name: 'group', id: 1) }
+  let(:type) { 'notifier' }
+  let(:recipient) { double(User, id: 2) }
+  let(:recipients) { [recipient] }
+  let(:new_notifications) { [double(Notification)] }
+  let(:client) { double('client') }
+  let(:mailer) { double('mailer') }
+  let(:data) do
+    {
+      user: current_user.name,
+      group_id: group.id,
+      group: group.name,
+      type: type,
+      uniqueid: "notifier_#{current_user.id}"
+    }.to_json
+  end
+  subject { GroupNotifier.new(group, type, current_user) }
+
+  it 'creates push notifications' do
+    allow(subject).to receive(:create_notification)
+    allow(subject).to receive(:send_email_notification)
+    allow(Notification).to receive_message_chain(:where, :order).and_return(new_notifications)
+    allow(Pusher).to receive(:[]).with("private-#{recipient.id}").and_return(client)
```

Comment:
Could this be `Array` instead of `:[]`?

Reply:
TBH I'm not sure, I'll give it a try

Reply:
Actually TIL so ruby access hashes using `.[]('key-here')` as a method:
![screenshot from 2018-10-04 20-44-46](https://user-images.githubusercontent.com/16313206/46510362-81f5b780-c816-11e8-83e3-2dd9f2a240d2.png)

Reply:
works the same for arrays as well